packages/cfd/src/features/Containers/top-up-demo-modal.tsx

```diff
@@ -29,9 +42,39 @@ const TopUpDemoModal = observer(({ platform }: TTopUpDemoModalProps) => {
         closeSuccessTopUpModal,
     } = ui;
 
-    const { current_account, dxtrade_companies, mt5_companies, topUpVirtual } = useCfdStore();
+    const { current_account } = useCfdStore();
+    const invalidate = useInvalidateQuery();
+    const { mutateAsync: MT5Deposit, status: mt5_deposit_status, error: mt5_deposit_error } = useMT5Deposit();
+    const {
+        mutateAsync: OtherCFDPlatformsDeposit,
+        status: other_cfd_deposit_status,
+        error: other_cfd_deposit_error,
+    } = useOtherCFDPlatformsDeposit();
 
     const ctrader_companies = getCTraderCompanies();
+    const dxtrade_companies = getDxCompanies();
+    const mt5_companies = getMtCompanies(is_eu);
+
+    useEffect(() => {
+        if (mt5_deposit_status === QUERY_STATUS.SUCCESS) invalidate('mt5_login_list');
```

Comment:
@shontzu-deriv The invalidation of a query should happen in the hook level, not component level

Reply:
Update the hooks file like this

<img width="703" alt="image" src="https://github.com/binary-com/deriv-app/assets/120543468/1e0c056f-b01b-4179-8e2a-ef9b5ac96073">